.golangci.yml

```diff
@@ -0,0 +1,153 @@
+# options for analysis running
```

Comment:
I think this is a template in the other repositories to allow substituting in at least the go-version from the `.go-version` file we haven't added to this repo yet.

We should probably do the same. Here's the commit that added linting to beats for example: https://github.com/elastic/beats/commit/0a749b54304a2fa7924c60129f05451e519496f5

To get CI to run it you'd just need to drop the github action in: https://github.com/elastic/elastic-agent-libs/blob/main/.github/workflows/golangci-lint.yml

We don't need to do the linter setup as part of this commit if it is adding too much work.

Reply:
Ah, that's interesting. I coped the out out of the `elastic-agent` repo, which didn't have the autogen warning.

Reply:
alright, copied in the CI files and fixed the file generation, not sure if there's anything else we need to do to make the linter happy.

Reply:
I think we also need to exclude generated go files (`*.pb.go`) from linting (see `skip-files` in the [docs](https://golangci-lint.run/usage/configuration/)). The change must be made in the template and then rendered via `mage linter:updategoversion`.

And we definitely need to try run the linter after the final config is rendered just to make sure the next PR would not fail (the current PR is not running the linter workflow yet).

Reply:
Alright, added the `skip-files` section. 

> And we definitely need to try run the linter after the final config is rendered

You mean the protobuf files? Right now the `mage` command that generates that isn't in CI, I assume we want it to be?

Reply:
1. Yes we will want the command to generate the protobuf files to run in CI, otherwise the CI build will fail when we turn it on.
2. @rdner is suggesting that we enable the linter as part of these changes. You can just create the github action to do this by copying one from our other repositories: https://github.com/elastic/elastic-agent-libs/blob/main/.github/workflows/golangci-lint.yml.